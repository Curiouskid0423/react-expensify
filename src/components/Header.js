import React from "react";

/**
 * All functional components also have to be Capitalized to be considered a
 * React component, whereas a class-based component needs to be Caps and extending
 * React.Component to be considered.
 */

const Header = (props) => {
    return (
        <div className = "header">
            <h1 className = "header__title">{props.title}</h1>
            <h2 className = "header__subtitle">{props.subtitle}</h2>
        </div>
    );
};

Header.defaultProps = {
    title: "Hello I am Kevin"
}

export default Header;