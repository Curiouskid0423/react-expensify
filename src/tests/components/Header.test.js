import React from "react";
import ReactShallowRenderer from "react-test-renderer/shallow";
import Header from "../../components/Header";

// Test 1. Test Header with snapshots. Shallow rendering
// works in contrast with Full Rendering with DOM.
test("Test with shallow rendering on Header.", () => {
   const renderer = new ReactShallowRenderer();
   renderer.render(<Header />);
   expect(renderer.getRenderOutput()).toMatchSnapshot();
});