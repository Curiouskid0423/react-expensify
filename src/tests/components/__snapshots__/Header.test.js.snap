// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the behavior of Header Component should shallow render a Header. 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <NavLink
    activeClassName="is-active"
    exact={true}
    to="/"
  >
     Home 
  </NavLink>
  <NavLink
    activeClassName="is-active"
    to="/create"
  >
     AddExpense 
  </NavLink>
  <button
    onClick={[MockFunction]}
  >
     LogOut 
  </button>
</header>
`;
